### ISSUE TITLE
Capytaine convention

### ISSUE THREAD

**Issue by cmichelenstrofer** [mancellin/capytaine#153]
Capytaine and Nemoh use the following convention for time-frequency domain:

$x(t) = R(Xe^{-i\omega t})$

This is different from what is done in most codes and textbooks (positive, rather than negative exponent):

$x(t) = R(Xe^{i\omega t})$

This often leads to confusion and bugs, and has been discussed in several issues (#41 #102 #110 #133 #149). 
I hope in this issue we can discuss options as well as document what would be affected by a change of convention. 

These are some of the options I can think of:

**Option 1: Keep current convention**
If the current convention is kept, I think the documentation should make the following clear/explicit:

1.  the current convention
2. its consequences (e.g. impedance phase  180 degrees)
3. how to convert the outputs to the + convention. 

**Option 2: Change convention**
I think the biggest differences between the two conventions is in the post-processing outputs (impedance, RAOs). I do not think any of the BEM solutions (added mass, radiation, complex excitation coefficients) would change (am I missing something?, what is the difference in wave convention #133?). But for consistency some of the code internals would need to be changed (@mancellin mentioned the relation between pressure and potential, and there are likely other). 

**Option 3: `convention` argument in post-processing functions**
Allow the user to specify the convention for any post-processing outputs. Keep all internals in the current convention. Document this.

**Comment by mancellin** [comment]
We clearly need to clear up and document the conventions used in Capytaine.
So thank you @cmichelenstrofer for starting this discussion.

I'm planning a breaking change release later this year. Changing this convention is an option, if there is a very good case for it.

> If the current convention is kept, I think the documentation should make the following clear/explicit

Whether we change it or not, the documentation is clearly missing a section on "Conventions and differences with other codes".

> I do not think any of the BEM solutions (added mass, radiation, complex excitation coefficients) would change

It might affect other outputs of the solver, such as the free surface elevation and the Kochin function.

> But for consistency some of the code internals would need to be changed (@mancellin mentioned the relation between pressure and potential, and there are likely other).

Now that I think about it: there is a time derivative in the boundary condition on the free surface, so this could affect the expression of the Green function.

> Option 3: convention argument in post-processing functions

I'm strongly against this option. As explained in #133, I believe handling datasets with different conventions within the same code is a source of headaches and errors (worse than the negative exponent of the current convention) for very little gain. The only place I would accept transforming data from one convention to another is at the "boundary" of the code (e.g. in an hypothetical `write_results_in_Wamit_file_format()` function).

**Comment by dtgaebe** [comment]
@mancellin 

I think the biggest reason for changing the convention is the interface to subsequent simulation environments, signal processing conventions (imagine using the admittance directly as transfer function for system analysis), and the somewhat physical interpretation of the standard Fourier transform convention, i.e. that the complex phasors of the Fourier trafo rotate clockwise, opposite to signal the complex amplitudes, assuming positive frequency signals and thus the multiplication of the signal frequency vector and the Fourier freq. vector cancels out the complex part, but the magnitudes add up.

**Comment by ryancoe** [comment]
I agree with @mancellin that having a `convention` argument is probably not a good idea. I would vote for **Option 2**, but of course explain this in the documentation and ideally tell/demonstrate how to switch the phase convention as a post-processing step (part of the cookbook maybe).

**Comment by mancellin** [comment]
In any case, the first step is to identify where and how this convention influences the code. I'll add that to my todo list.

**Comment by mancellin** [comment]
> Now that I think about it: there is a time derivative in the boundary condition on the free surface, so this could affect the expression of the Green function.

Just leaving a note to myself: the boundary condition is a second-order derivative, so it is not impacted by the convention.

**Comment by salhus** [comment]
I recently took a look at this issue for WEC-Sim, where a MATLAB script imports hydrodynamics coefficients from different BEM platforms.
The way i addressed this was add a phase of $\pi$ in the argument of the sinusoidal function. Initially I approached this by simply adding a negative sign but it doesnt work for all situations for the reasons identified above.
If you are still in the process of resolving this, perhaps this PR might be of some relevance.
https://github.com/WEC-Sim/WEC-Sim/pull/855

**Comment by mancellin** [comment]
Just adding a piece of trivia on when the minus sign makes sense.

Nemoh/Capytaine's convention can be written as

$$ x(t) = \Re(\hat x) \cos(\omega t) + \Im(\hat x) \sin(\omega t), $$

whereas WAMIT's convention can be written as

$$ x(t) = \Re(\hat x) \cos(\omega t) - \Im(\hat x) \sin(\omega t). $$

Capytaine's convention is more natural when decomposing in $\cos$ and $\sin$ (as it is done e.g. in Delhommeau's thesis), while WAMIT's convention is more straightforward when working with complex numbers.

---

Since this thread was opened, the "Conventions and differences with other codes" has been extended and as new features got implemented, changing the convention becomes more and more costly.
The topic of exporting the data in other format/convention and interfacing with other software is still largely open.
